src/code/z_vr_box.c

```diff
@@ -121,11 +553,12 @@ void Skybox_Setup(GlobalContext* globalCtx, SkyboxContext* skyboxCtx, s16 skybox
 
             DmaMgr_SendRequest1(skyboxCtx->staticSegments[1], D_8011FD3C[sp40].unk_0, size, "../z_vr_box.c", 1064);
 
-            // reorderings in the rest of this case
+            if (((sp41 & 4) >> 2) != (sp41 & 1)) {
+                D_8011FD3C[sp41].unk_C = D_8011FD3C[sp41].unk_C; // fake but greatly improves match.
+
+                size = (D_8011FD3C[sp41].unk_C - D_8011FD3C[sp41].unk_8) * 2;
 
-            if (((u8)(sp41 & 4) >> 2) != (sp41 & 1)) {
-                size = D_8011FD3C[sp41].unk_C - D_8011FD3C[sp41].unk_8;
-                skyboxCtx->staticSegments[2] = GameState_Alloc(&globalCtx->state, size * 2, "../z_vr_box.c", 1072);
+                skyboxCtx->staticSegments[2] = GameState_Alloc(&globalCtx->state, size, "../z_vr_box.c", 1072);
```

Comment:
All `skyboxCtx->staticSegments[2]` accesses should be updated to the palettes member (2 is out of bounds now)